syntax = "proto3";

package narrator.templates;

import "common.proto";

service TemplateService {
  rpc GetTemplate (GetTemplateRequest) returns (narrator.common.Template);
  rpc FindTemplates (FindTemplatesRequest) returns (FindTemplatesResponse);
  rpc UpsertTemplate (UpsertTemplateRequest) returns (narrator.common.Template);
}

message GetTemplateRequest {
  string template_id = 1;
}

message FindTemplatesRequest {
  string query = 1;
  string category = 2;
  uint32 top_k = 3;
}

message FindTemplatesResponse {
  repeated narrator.common.Template templates = 1;
}

message UpsertTemplateRequest {
  narrator.common.Template template = 1;
}
